Mathlib/Combinatorics/HalesJewett.lean

```diff
@@ -67,6 +67,15 @@ combinatorial line, Ramsey theory, arithmetic progression
 
 -/
 
+namespace Equiv
+variable {α β : Type*}
+
+protected lemma exists_congr {p : α → Prop} {q : β → Prop} (f : α ≃ β) (h : ∀ x, p x ↔ q (f x)) :
+    (∃ x, p x) ↔ ∃ y, q y :=
+  f.exists_congr_left.trans $ by simp [h]
```

Comment:
I find it quite hard to believe this doesn't already exist.

Reply:
Me too, but we do have the `existsUnique` and `forall` versions in `Logic.Equiv.Defs`.